libnmstate/ifaces/linux_bridge.py

```diff
@@ -0,0 +1,250 @@
+        since it enables the user to simply specify the updated values rather
+        than the full current state + the updated value.
+        """
+        if not self._ports_config:
+            # User explictly defined empty list for ports and expecting
+            # removal of all ports.
+            return
+
+        other_indexed_ports = _index_ports_config(other._ports_config)
+        self_indexed_ports = _index_ports_config(self._ports_config)
+
+        # When defined, user need to specify the whole list of ports.
+        for port_iface_name in (
+            other_indexed_ports.keys() & self_indexed_ports.keys()
+        ):
+            merge_dict(
+                self_indexed_ports[port_iface_name],
+                other_indexed_ports[port_iface_name],
+            )
+        self.raw[LinuxBridge.CONFIG_SUBTREE][LinuxBridge.PORT_SUBTREE] = list(
+            self_indexed_ports.values()
+        )
+
+    def _fix_vlan_filering_mode(self):
+        for port_config in self._ports_config:
+            _vlan_config_clean_up(_get_port_vlan_config(port_config))
+
+    def state_for_verify(self):
+        state = super().state_for_verify()
+        _normalize_linux_bridge_port_vlan(state)
+        return state
+
+    def validate(self):
+        super().validate()
+        self._validate_vlan_filtering_trunk_tags()
+        self._validate_vlan_filtering_tag()
+        self._validate_vlan_filtering_enable_native()
+
+    def _validate_vlan_filtering_trunk_tags(self):
+        for port_config in self.original_dict.get(
+            LinuxBridge.PORT_SUBTREE, []
+        ):
+            port_vlan_state = port_config.get(
+                LinuxBridge.Port.VLAN_SUBTREE, {}
+            )
+            vlan_mode = port_vlan_state.get(LinuxBridge.Port.Vlan.MODE)
+            trunk_tags = port_vlan_state.get(
+                LinuxBridge.Port.Vlan.TRUNK_TAGS, []
+            )
+
+            if vlan_mode == LinuxBridge.Port.Vlan.Mode.ACCESS:
+                if trunk_tags:
+                    raise NmstateValueError(
+                        "Access port cannot have trunk tags"
+                    )
+            elif port_vlan_state:
+                if not trunk_tags:
+                    raise NmstateValueError(
+                        "A trunk port needs to specify trunk tags"
+                    )
+            for trunk_tag in trunk_tags:
+                _assert_vlan_filtering_trunk_tag(trunk_tag)
+
+    def _validate_vlan_filtering_tag(self):
+        """
+        The "tag" is valid in access mode or tunk mode with
+        "enable-native:True".
+        """
+        for port_config in self.original_dict.get(
+            LinuxBridge.PORT_SUBTREE, []
+        ):
+            vlan_config = _get_port_vlan_config(port_config)
+            if (
+                vlan_config.get(LinuxBridge.Port.Vlan.TAG)
+                and _vlan_is_trunk_mode(vlan_config)
+                and _vlan_is_enable_native(vlan_config)
```

Comment:
Isn't this condition actually OK ? If we're on trunk mode, *and* have native vlan enabled , the `tag` attribute is a valid config.

Reply:
Changed to

```python
            if (
                vlan_config.get(LinuxBridge.Port.Vlan.TAG)
                and _vlan_is_trunk_mode(vlan_config)
                and not _vlan_is_enable_native(vlan_config)
            ):
                raise NmstateValueError(
                    "Tag cannot be use in trunk mode without enable-native"
                )
```